generators/factoryMethod/index.js

```diff
@@ -1,81 +1,79 @@
 'use strict';
 
+/**
+ * Adds a method to a previously generated factory.
+ */
+
 var Generator = require('yeoman-generator'),
     chalk     = require('chalk'),
-    yosay     = require('yosay'),
-    fs        = require('fs');
+    caseIt    = require('change-case'),
+    fsp       = require('fs-promise');
 
 module.exports = class extends Generator {
+  // Prompting Queue
   prompting () {
-
-    var prompts = [{
+    const prompts = [{
       type: 'input',
       name: 'factoryName',
       message: 'What is the name of the Factory?',
+      validate: (answer) => {
+        let prefix   = this.config.get('prefix') ? this.config.get('prefix') : 'app',
+            filename = caseIt.paramCase(`${prefix}-${answer}`),
+            filepath = this.destinationPath(`src/factories/${filename}/${filename}.factory.js`);
+
+        if (this.fs.exists(filepath))
+          return true;
+        else
+          return `Factory ${answer} does not exist.`;
+      }
     }, {
       type: 'input',
       name: 'methodName',
-      message: 'What is the name of the Method',
+      message: 'Name of this method?',
+      validate: Boolean
     }, {
       type: 'input',
       name: 'methodDescription',
-      message: 'Describe this method',
+      message: 'Describe this method.',
+      default: 'description'
     }];
 
-    return this.prompt(prompts).then(function (props) {
-      // To access props later use this.props.someAnswer;
-      this.props = props;
-
-    }.bind(this));
+    return this.prompt(prompts)
+      .then((answers) => {
+        this.props          = answers;
+        this.props.prefix   = this.config.get('prefix') ? this.config.get('prefix') : 'app';
+        this.props.filename = caseIt.paramCase(`${this.props.prefix}-${this.props.factoryName}`);
+        this.props.filepath = `src/factories/${this.props.filename}/${this.props.filename}.factory.js`;
+        this.props.method   = caseIt.camelCase(this.props.methodName);
+      });
   }
 
+  // Writing Queue
   writing () {
-
-    var properties = {
-      description: this.props.methodDescription,
-      factory: this.props.factoryName.toLowerCase(),
-      method: this.props.methodName.replace(/(\s|[^A-Za-z0-9])+./g, function(match){
-        return match.slice(match.length-1, match.length).toUpperCase();
-      }).replace(/[^A-Za-z0-9]+$/, ""),
-      prefix: this.config.get('prefix') ? this.config.get('prefix') + '-' : 'app' +'-',
-    };
-
-    if(properties.factory.slice(0, properties.prefix.length) === properties.prefix) {
-      properties.fileName = properties.factory;
-    }else{
-      properties.fileName = properties.prefix + properties.factory;
-    }
```

Comment:
I've dropped this prefix check here, presuming that a user will not try to add the prefix to the factory name. If, during use, it becomes clear that we need to put this back in I'm for it. I just wanted to simplify and I am trying to give users credit for being smart enough to leave off the prefix since it is automatically handled throughout the other generators.